designer/test/component-create.test.js

```diff
@@ -4,15 +4,22 @@ import * as Code from '@hapi/code'
 import * as Lab from '@hapi/lab'
 import ComponentCreate from '../client/component-create'
 import { Data } from '../client/model/data-model'
+import sinon from 'sinon'
 
 const { expect } = Code
 const lab = Lab.script()
 exports.lab = lab
-const { suite, test } = lab
+const { before, suite, test } = lab
 
 suite('Component create', () => {
   const data = new Data({})
   const page = { path: '/1' }
+  const generatedId = 'DMaslknf'
+
+  before(() => {
+    data.getId = sinon.stub()
+    data.getId.resolves(generatedId)
```

Comment:
I think you can one line this `sinon.stub().resolves(generatedId)`

Reply:
Fixed